src/dotnet/commands/dotnet-build3/MSBuildForwardingApp.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.InternalAbstractions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class MSBuildForwardingApp : ForwardingApp
+    {
+        private static readonly string s_msbuildExeName = "MSBuild.exe";
+
+        public MSBuildForwardingApp(string[] argsToForward) :
+            base(GetMSBuildExePath(), 
+                argsToForward,
+                depsFile: GetDepsFile(),
+                runtimeConfig: GetRuntimeConfig(),
+                environment: GetEnvironment()) { }
+
+        private static Dictionary<string, string> GetEnvironment()
+        {
+            return new Dictionary<string, string>
+            {
+                { "MSBuildExtensionsPath", GetDotnetMSBuildTargetsPath() },
+                { "DotnetHostPath", GetHostPath() },
+                { "BaseNuGetRuntimeIdentifier", GetCurrentBaseRid() }
+            };
+        }
+
+        private static string GetCurrentBaseRid()
+        {
+            return RuntimeEnvironment.GetRuntimeIdentifier()
+                .Replace("-" + RuntimeEnvironment.RuntimeArchitecture, "");
+        }
+
+        private static string GetHostPath()
+        {
+            return new Muxer().MuxerPath;
+        }
+
+        private static string GetDotnetMSBuildTargetsPath()
+        {
+            return Path.Combine(AppContext.BaseDirectory);
```

Comment:
Any reason that you Path.Combine only 1 string?  Why not just return AppContext.BaseDirectory?